CommDLL.pas

```diff
@@ -1,46 +1,46 @@
-unit CommDLL;
-interface
+unit CommDLL;
+interface
 uses
   Windows, System.SysUtils, ExtAIAPI, InterfaceDelphi, CommonDataTypes;
-
-const
-  MAX_HANDS = 12;
-
-type
-  TInitDLL = procedure(); StdCall;
-  TTerminDLL = procedure(); StdCall;
-  TInitNewExtAI = procedure(aID: ui8; aActions: IActions) StdCall;
-  TNewExtAI = function(): IEvents; SafeCall; // Same like StdCall but allows exceptions
-
-  TCallback1 = function(sig: ui8): b of object; StdCall;
-
-  // Communication with 1 physical DLL with using exported methods.
-  // Main targets: initialization of DLL, creation of ExtAIs and termination of DLL and ExtAIs
-  TCommDLL = class
-  private
-    fLibHandle: THandle;
-    fDLLPath: String;
-    fExtAIAPICnt: ui8;
-    fExtAIAPI: array [0..MAX_HANDS-1] of TExtAIAPI;
-    fOnInitDLL: TInitDLL;
+
+const
+  MAX_HANDS = 12;
+
+type
+  TInitDLL = procedure(); StdCall;
+  TTerminDLL = procedure(); StdCall;
+  TInitNewExtAI = procedure(aID: ui8; aActions: IActions) StdCall;
+  TNewExtAI = function(): IEvents; SafeCall; // Same like StdCall but allows exceptions
+
+  TCallback1 = function(sig: ui8): b of object; StdCall;
+
+  // Communication with 1 physical DLL with using exported methods.
+  // Main targets: initialization of DLL, creation of ExtAIs and termination of DLL and ExtAIs
+  TCommDLL = class
+  private
+    fLibHandle: THandle;
+    fDLLPath: String;
+    fExtAIAPICnt: ui8; //@Martin: Use Delphi datatypes in Delphi please. Integer here
+    fExtAIAPI: array [0..MAX_HANDS-1] of TExtAIAPI; //@Martin: Better change to dynamic array or TList<> to avoid MAX_HANDS const (it is not needed here)
+    fOnInitDLL: TInitDLL;
     fOnTerminDLL: TTerminDLL;
     fOnInitNewExtAI: TInitNewExtAI;
-    fOnNewExtAI: TNewExtAI;
-  public
-    property DLLPath: String read fDLLPath;
-    //property ExtAICnt: ui8 read fExtAIAPICnt;
-    //property ExtAI: TExtAI read fExtAI
-
-    constructor Create();
-    destructor Destroy(); override;
-
-    function LinkDLL(aDLLPath: String): b;
-    function CreateNewExtAI(aExtAIID: ui8): b;
-    procedure UpdateState();
-
-    function Callback1(sig: ui8): b; StdCall;
-  end;
-
+    fOnNewExtAI: TNewExtAI;
+  public
+    property DLLPath: String read fDLLPath;
+    //property ExtAICnt: ui8 read fExtAIAPICnt;
+    //property ExtAI: TExtAI read fExtAI
+
+    constructor Create();
+    destructor Destroy(); override;
+
+    function LinkDLL(aDLLPath: String): b;
+    function CreateNewExtAI(aExtAIID: ui8): b; //@Martin: This should return something (through TKMExtAIMaster maybe?). Something that TKMExtAI will be able to use directly

```

Comment:
DLL and each ExtAI must to return at least its configuration when they are created. DLL for the TCommDLL and ExtAI for ExtAIAPI. To the game we can return name of ExtAI etc. However, with exception of message that ExtAI was created, I dont see any important message which should be implemented in first version.

Reply:
Let's continue that in he DPR backbone discussion )